### DISCUSSION TITLE
請問可否可透過什麼配置，讓登入帳號無法下載工單詳情？

### DISCUSSION THREAD

**Discussion by dearsiang** [hhyo/Archery#2115] [Q&A]
Hi 你好

這邊看到工單詳情可以下載成多種格式的檔案。

請問可否可透過什麼配置，讓登入帳號無法下載工單詳情？

**Comment by LeoQuote** [comment]
只要在给用户授权时, 注意不要给 "在线查询下载权限" 这个权限就可以了

**Reply by dearsiang** [reply]
Hi Leo

這邊使用預設RD"組"的方式，給予帳號權限，這邊檢查RD組，沒有在線查詢下載權限，但在RD組的帳號，還是可以下載工單詳情

**Reply by dearsiang** [reply]
不知道你有沒有其他聯絡方式，這邊可以用貼圖方式，讓你更加瞭解問題。

**Reply by LeoQuote** [reply]
你现在的版本是什么? 可以考虑先更新一下版本到最新再试试

**Reply by LeoQuote** [reply]
有可能你使用的版本系统中就没有这个权限, 也没有和这个权限相关的控制逻辑.

**Reply by dearsiang** [reply]
這邊使用Archery v1.9.1，應是最新版本

**Reply by LeoQuote** [reply]
1.9.1 应该已经有了权限了.

https://github.com/hhyo/Archery/blob/813a16e37fa0ad63eaa3de80ceed11522d41e6a7/sql/views.py#L315-L330

看一下这边的逻辑, 你可以尝试再检查一下用户或者用户组是否为超级用户

**Reply by dearsiang** [reply]
Hi Leo

這邊檢查用戶組跟用戶，都不是超級用戶

**Reply by dearsiang** [reply]
<img width="314" alt="Snipaste_2023-04-07_11-19-36" src="https://user-images.githubusercontent.com/107904775/230558975-d46ab4e8-a28e-4270-bdd5-b7821e36e343.png">
<img width="228" alt="Snipaste_2023-04-07_11-19-23" src="https://user-images.githubusercontent.com/107904775/230559001-e0dd50da-2607-4e94-af76-1729e7da4375.png">

這是目前RD組的權限清單

**Reply by dearsiang** [reply]
Hi Leo

請問這是Bug 對嗎？

**Reply by LeoQuote** [reply]
我不确认这是不是 bug, 你能否尝试一下我们master 分支上的代码呢? 有可能在master 分支中, 就没有这个问题了

**Reply by dearsiang** [reply]
這邊是用 master 分支去建置系統的，還是要退版呢？

**Reply by dearsiang** [reply]
這邊退版回add_init_sql 還是一樣的情況
![Uploading Snipaste_2023-04-10_09-28-50.png…]()

**Reply by dearsiang** [reply]
另外參考ssl 配置指南步驟，配置後，服務不能正常運作

**Reply by LeoQuote** [reply]
我不确认你说的是什么意思, 你说的意思是, 用 master 分支的镜像, 服务不正常是吗?  你能详细说明一下你的配置步骤吗? 你有可能使用了master 分支下的 docker-compose 文件, 但是没有使用最新的 python 代码, 最好是你自己来构建镜像, 然后启动服务, 这个是能确认在使用最新的代码的

**Comment by dearsiang** [comment]
Hi Leo

這邊使用https://archerydms.com/installation/docker/，這個進行安裝，但還是可以下載工單詳情

請問要如何自已勾建鏡像？

**Comment by dearsiang** [comment]
Hi Leo

請問"在线查询下载权限"，這個權限是在附件圖紅色標示處得功能權限限制嗎？
<img width="1440" alt="Snipaste_2023-04-10_16-47-08" src="https://user-images.githubusercontent.com/107904775/230867411-21f8f1a2-3a4d-4115-aba8-ede0a5b058c7.png">

**Reply by LeoQuote** [reply]
实在不好意思, 原来我误解了你的意思, 原来你的意思是这里的下载按钮, 很抱歉这部分的下载按钮并没有相关选项可以控制, 欢迎pr.

另外关于禁用下载, 我们这边的意见一直都是, 禁用下载意义不大, 更多的是一种象征意义? 如果用户想要保存, 依然可以使用截图, 摄屏等方式保存信息, 而且这些方法的使用门槛都不高.
